internal/controller/defaults.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"os"
+
+	csiv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/ceph/ceph-csi-operator/internal/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+)
+
+var imageDefaults = map[string]string{
+	"provisioner": "registry.k8s.io/sig-storage/csi-provisioner:v5.0.1",
+	"attacher":    "registry.k8s.io/sig-storage/csi-attacher:v4.6.1",
+	"resizer":     "registry.k8s.io/sig-storage/csi-resizer:v1.11.1",
+	"snapshotter": "registry.k8s.io/sig-storage/csi-snapshotter:v8.0.1",
+	"registrar":   "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.10.1",
+	"plugin":      "quay.io/cephcsi/cephcsi:v3.10.2",
+	"addons":      "quay.io/csiaddons/k8s-sidecar:v0.8.0",
+}
+
+const (
+	defaultGRrpcTimeout   = 150
+	defaultSnapshotPolicy = csiv1a1.AutoDetectSnapshotPolicy
+	defaultKubeletDirPath = "/var/lib/kubelet"
+)
+
+var defaultLeaderElection = csiv1a1.LeaderElectionSpec{
+	LeaseDuration: 137,
+	RenewDeadline: 107,
+	RetryPeriod:   26,
+}
+
+var defautUpdateStrategy = appsv1.DaemonSetUpdateStrategy{
+	Type: appsv1.RollingUpdateDaemonSetStrategyType,
+	RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+		MaxUnavailable: ptr.To(intstr.FromInt(1)),
+	},
+}
+
+var operatorNamespace = (func() string {
```

Comment:
The function needs to be invoked immediately. 
Enclosing it removes any ambiguity regarding what is being invoked, which means it is more readable.

Reply:
ack, I understand what you were doing, however I didn't observe that you were calling it and so raised a comment.

Nevertheless, I don't think the latest refactor `util.Call` isn't adding any functionality (or I couldn't grasp), original was good enough.

Reply:
@leelavg As you already saw in your original review it was not obvious that the function is being called. That usually happens with longer tasks that do not fit in a single screen height. The main issue is that the indicator for invocation is at the end of the function definition. 

The `Call` util does not add anything from a functionality perspective but it does resolve the reader's potential false ambiguity because it moves the intention (of executing the function in place) to the same line of the function definition.

Reply:
> does not add anything from a functionality perspective but it does resolve the reader's potential
- my initial comment has nothing to do with this, I just mentioned `var ns = func() string{ body }()` is not too big to comprehend

> that do not fit in a single screen height
- anyways, agreeing to this, can be resolved.